proxy.go

```diff
@@ -197,8 +198,20 @@ func executeWithRetry(
 	startTime := time.Now()
 	var since float64
 
+	// keep the request body
+	body, err := ioutil.ReadAll(req.Body)
+	if err != nil {
+		since = time.Since(startTime).Seconds()
+
+		return since, err
+	}
+	req.Body.Close()
+
 	numRetry := 0
 	for {
+		// update body
+		req.Body = ioutil.NopCloser(bytes.NewBuffer(body))
```

Comment:
don't you need to close this req.Body at some point in the loop? (If not, why doing the req.Body.Close() line 208?

Reply:
Yes, you are right, I need to close the body of the request. I change the code and use defer req.Body.Close(). The request Body needs to be closed immediately after reading it.

Reply:
I don't think you should defer it because you're in the infinite loop that can override multiple times req.Body

Reply:
Ok, since the defer is executed when the surrounding function returns, it will not execute when the body is updated. I will close the req.Body after reading or updating it.